File: android/src/main/kotlin/io/sourcya/playx_3d_scene/core/models/scene/Scene.kt
package io.sourcya.playx_3d_scene.core.models.scene

import io.sourcya.playx_3d_scene.core.models.scene.camera.Camera
import io.sourcya.playx_3d_scene.core.models.scene.light.IndirectLight
import io.sourcya.playx_3d_scene.core.models.scene.shapes.Ground
import io.sourcya.playx_3d_scene.utils.toObject

data class Scene(
    val skybox: Skybox? = null,
    var light: Light?= null,
    var indirectLight: IndirectLight? = null,
    val camera: Camera? = null,
    val ground: Ground? = null,
) {

    companion object {
        fun fromMap(map: Map<String?, Any?>?): Scene? {
            return map?.toObject<Scene>()
        }

    }

}

File: lib/models/scene/scene.dart
import 'package:playx_3d_scene/models/scene/camera/camera.dart';
import 'package:playx_3d_scene/models/scene/ground.dart';
import 'package:playx_3d_scene/models/scene/indirect_light/indirect_light.dart';
import 'package:playx_3d_scene/models/scene/light/light.dart';
import 'package:playx_3d_scene/models/scene/skybox/skybox.dart';

class Scene {
  Skybox? skybox;
  IndirectLight? indirectLight;
  Light? light;
  Camera? camera;
  Ground? ground;

  Scene(
      {this.skybox, this.indirectLight, this.light, this.camera, this.ground});

  Map<String, dynamic> toJson() => {
        'skybox': skybox?.toJson(),
        'light': light?.toJson(),
        'indirectLight': indirectLight?.toJson(),
        'camera': camera?.toJson(),
        'ground': ground?.toJson(),
      };
}
